Vanilla rendering has dependency to material-ui

When I try to follow the tutorial, but replacing material-renderers with @jsonforms/vanilla-renderers there's a dependency to material-ui in the jsonforms core:
 
./node_modules/@jsonforms/react/lib/DispatchField.js
Module not found: Can't resolve '@material-ui/core' in 'jsonforms/node_modules/@jsonforms/react/lib'

--- a/packages/core/src/util/field.ts
+++ b/packages/core/src/util/field.ts
@@ -146,9 +146,7 @@ export const mapStateToDispatchFieldProps = (
   };
 };
 
-export interface DispatchFieldProps extends DispatchFieldStateProps {
-  showError: boolean;
-}
+export interface DispatchFieldProps extends DispatchFieldStateProps {}
 
 /**
  * Default mapStateToFieldProps for enum field. Options is used for populating dropdown list

--- a/packages/material/src/complex/MaterialTableControl.tsx
+++ b/packages/material/src/complex/MaterialTableControl.tsx
@@ -22,19 +22,30 @@
   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
   THE SOFTWARE.
 */
+import isEmpty from 'lodash/isEmpty';
+import filter from 'lodash/filter';
+import { DispatchField } from '@jsonforms/react';
 import capitalize from 'lodash/capitalize';
-import IconButton from '@material-ui/core/IconButton';
 import React from 'react';
-import { Table, TableBody, TableCell, TableHead, TableRow, Typography } from '@material-ui/core';
+import {
+  FormHelperText,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Typography
+} from '@material-ui/core';
 import {
   ArrayControlProps,
   ControlElement,
+  formatErrorMessage,
   Generate,
   Helpers,
   JsonSchema,
   Paths
 } from '@jsonforms/core';
-import { DispatchField } from '@jsonforms/react';
+import IconButton from '@material-ui/core/IconButton';
 import DeleteIcon from '@material-ui/icons/Delete';
 import { WithDeleteDialogSupport } from './DeleteDialog';
 import { ErrorObject } from 'ajv';
@@ -60,35 +71,38 @@ const generateCells = (
 ) => {
 
   if (scopedSchema.type === 'object') {
-    return getValidColumnProps(scopedSchema)
-      .map(prop => {
-        const cellPath = Paths.compose(rowPath, prop);
-        const props = {
-          propName: prop,
-          scopedSchema,
-          rowPath,
-          cellPath,
-          errors: cellErrors
-        };
+    return getValidColumnProps(scopedSchema).map(prop => {
+      const cellPath = Paths.compose(
+        rowPath,
+        prop
+      );
+      const props = {
+        propName: prop,
+        scopedSchema,
+        rowPath,
+        cellPath,
+        errors: cellErrors
+      };
 
-        return (<Cell key={cellPath} {...props} />);
-      });
+      return <Cell key={cellPath} {...props} />;
+    });
   } else {
     // primitives
     const props = {
       scopedSchema,
       rowPath,
       cellPath: rowPath,
-      errors: cellErrors,
+      errors: cellErrors
     };
-    return (<Cell key={rowPath} {...props} />);
+    return <Cell key={rowPath} {...props} />;
   }
 };
 
 const getValidColumnProps = (scopedSchema: JsonSchema) => {
   if (scopedSchema.type === 'object') {
-    return Object.keys(scopedSchema.properties)
-      .filter(prop => scopedSchema.properties[prop].type !== 'array');
+    return Object.keys(scopedSchema.properties).filter(
+      prop => scopedSchema.properties[prop].type !== 'array'
+    );
   }
   // primitives
   return [''];
@@ -118,9 +132,21 @@ interface NonEmptyCellProps {
   rowPath: string;
   propName?: string;
   scopedSchema: JsonSchema;
+  errors?: any[];
 }
 
-const NonEmptyCell = ({ rowPath, propName, scopedSchema }: NonEmptyCellProps) => (
+const NonEmptyCell = ({
+  rowPath,
+  propName,
+  scopedSchema,
+  errors
+}: NonEmptyCellProps) => {
+  const errorsPerEntry: any[] = filter(
+    errors,
+    error => error.dataPath === rowPath + '.' + propName
+  ).map(e => e.message);
+  const isValid = isEmpty(errorsPerEntry);
+  return (
     <React.Fragment>
       <NoBorderTableCell>
         <DispatchField
@@ -130,11 +156,14 @@ const NonEmptyCell = ({ rowPath, propName, scopedSchema }: NonEmptyCellProps) =>
             scopedSchema.type === 'object' ? `#/properties/${propName}` : '#'
           )}
           path={rowPath}
-          showError
         />
+        <FormHelperText error={!isValid}>
+          {!isValid && formatErrorMessage(errorsPerEntry)}
+        </FormHelperText>
       </NoBorderTableCell>
     </React.Fragment>
   );
+};
 
 interface NonEmptyRowProps {
   childPath: string;
@@ -143,22 +172,17 @@ interface NonEmptyRowProps {
   rowData: any;
 }
 
-const NonEmptyRow = (
-  {
-    childPath,
-    scopedSchema,
-    childErrors,
-    rowData,
-    openDeleteDialog
-  }: NonEmptyRowProps & WithDeleteDialogSupport) => (
-
-  <TableRow
-    key={childPath}
-    hover
-  >
+const NonEmptyRow = ({
+  childPath,
+  scopedSchema,
+  childErrors,
+  rowData,
+  openDeleteDialog
+}: NonEmptyRowProps & WithDeleteDialogSupport) => (
+  <TableRow key={childPath} hover>
     {generateCells(NonEmptyCell, scopedSchema, childPath, childErrors)}
     <NoBorderTableCell style={styles.fixedCell}>
-      <div style={{ display: 'flex', justifyContent: 'center'}}>
+      <div style={{ display: 'flex', justifyContent: 'center' }}>
         <IconButton
           aria-label={`Delete`}
           onClick={() => openDeleteDialog(childPath, rowData)}
@@ -197,9 +221,10 @@ const TableRows = (
   });
 };
 
-export class MaterialTableControl
-  extends React.Component<ArrayControlProps & WithDeleteDialogSupport, any> {
-
+export class MaterialTableControl extends React.Component<
+  ArrayControlProps & WithDeleteDialogSupport,
+  any
+> {
   render() {
     const {
       label,

--- a/packages/material/src/controls/MaterialInputControl.tsx
+++ b/packages/material/src/controls/MaterialInputControl.tsx
@@ -1,7 +1,7 @@
 /*
   The MIT License
 
-  Copyright (c) 2018 EclipseSource Munich
+  Copyright (c) 2018-2019 EclipseSource Munich
   https://github.com/eclipsesource/jsonforms
 
   Permission is hereby granted, free of charge, to any person obtaining a copy
@@ -57,19 +57,23 @@ export class MaterialInputControl extends Control<ControlProps, ControlState> {
     } = this.props;
     const isValid = errors.length === 0;
     const trim = config.trim;
-    const style: {[x: string]: any} = {};
+    const style: { [x: string]: any } = {};
     if (!visible) {
       style.display = 'none';
     }
-    const inputLabelStyle: {[x: string]: any} = {
+    const inputLabelStyle: { [x: string]: any } = {
       whiteSpace: 'nowrap',
-      overflow : 'hidden',
+      overflow: 'hidden',
       textOverflow: 'ellipsis',
       // magic width as the label is transformed to 75% of its size
       width: '125%'
     };
 
-    const showDescription = !isDescriptionHidden(visible, description, this.state.isFocused);
+    const showDescription = !isDescriptionHidden(
+      visible,
+      description,
+      this.state.isFocused
+    );
     return (
       <FormControl
         style={style}
@@ -78,18 +82,25 @@ export class MaterialInputControl extends Control<ControlProps, ControlState> {
         onBlur={this.onBlur}
         id={id}
       >
-        <InputLabel htmlFor={id + '-input'} error={!isValid} style={inputLabelStyle}>
+        <InputLabel
+          htmlFor={id + '-input'}
+          error={!isValid}
+          style={inputLabelStyle}
+        >
           {computeLabel(isPlainLabel(label) ? label : label.default, required)}
         </InputLabel>
         <DispatchField
-            uischema={uischema}
-            schema={schema}
-            path={path}
-            id={id + '-input'}
-            showError={false}
+          uischema={uischema}
+          schema={schema}
+          path={path}
+          id={id + '-input'}
         />
         <FormHelperText error={!isValid}>
-          {!isValid ? formatErrorMessage(errors) : showDescription ? description : null}
+          {!isValid
+            ? formatErrorMessage(errors)
+            : showDescription
+            ? description
+            : null}
         </FormHelperText>
       </FormControl>
     );

--- a/packages/react/src/DispatchField.tsx
+++ b/packages/react/src/DispatchField.tsx
@@ -1,7 +1,7 @@
 /*
   The MIT License
 
-  Copyright (c) 2018 EclipseSource Munich
+  Copyright (c) 2018-2019 EclipseSource Munich
   https://github.com/eclipsesource/jsonforms
 
   Permission is hereby granted, free of charge, to any person obtaining a copy
@@ -27,45 +27,36 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { UnknownRenderer } from './UnknownRenderer';
 import {
-  DispatchFieldProps, DispatchFieldStateProps,
-  formatErrorMessage,
-  mapStateToDispatchFieldProps, OwnPropsOfField,
+  DispatchFieldProps,
+  DispatchFieldStateProps,
+  mapStateToDispatchFieldProps,
+  OwnPropsOfField
 } from '@jsonforms/core';
-import { FormHelperText } from '@material-ui/core';
 
 /**
  * Dispatch renderer component for fields.
  */
 class Dispatch extends React.Component<DispatchFieldProps, any> {
-    render() {
-        const { uischema, schema, path, fields, id, errors, isValid, showError } = this.props;
-        const field = maxBy(fields, r => r.tester(uischema, schema));
+  render() {
+    const { uischema, schema, path, fields, id } = this.props;
+    const field = maxBy(fields, r => r.tester(uischema, schema));
 
-        if (field === undefined || field.tester(uischema, schema) === -1) {
-            return <UnknownRenderer type={'field'}/>;
-        } else {
-            const Field = field.field;
+    if (field === undefined || field.tester(uischema, schema) === -1) {
+      return <UnknownRenderer type={'field'} />;
+    } else {
+      const Field = field.field;
 
-            return (
-                <React.Fragment>
-                    <Field
-                        uischema={uischema}
-                        schema={schema}
-                        path={path}
-                        id={id}
-                    />
-                    {
-                        showError &&
-                        <FormHelperText error={!isValid}>
-                            {!isValid && formatErrorMessage(errors)}
-                        </FormHelperText>
-                    }
-                </React.Fragment>
-            );
-        }
+      return (
+        <React.Fragment>
+          <Field uischema={uischema} schema={schema} path={path} id={id} />
+        </React.Fragment>
+      );
     }
+  }
 }
 
-export const DispatchField = connect<DispatchFieldStateProps, {}, OwnPropsOfField>(
-  mapStateToDispatchFieldProps
-)(Dispatch);
+export const DispatchField = connect<
+  DispatchFieldStateProps,
+  {},
+  OwnPropsOfField
+>(mapStateToDispatchFieldProps)(Dispatch);

--- a/packages/vanilla/src/complex/TableArrayControl.tsx
+++ b/packages/vanilla/src/complex/TableArrayControl.tsx
@@ -47,17 +47,9 @@ import { addVanillaControlProps } from '../util';
 import { connect } from 'react-redux';
 import { VanillaRendererProps } from '../index';
 
-const {
-  createLabelDescriptionFrom,
-  convertToValidClassName
-} = Helpers;
+const { createLabelDescriptionFrom, convertToValidClassName } = Helpers;
 
-const {
-  or,
-  isObjectArrayControl,
-  isPrimitiveArrayControl,
-  rankWith
-} = Test;
+const { or, isObjectArrayControl, isPrimitiveArrayControl, rankWith } = Test;
 
 /**
  * Alternative tester for an array that also checks whether the 'table'
@@ -65,12 +57,14 @@ const {
  * @type {RankedTester}
  */
 export const tableArrayControlTester: RankedTester = rankWith(
-    3,
-    or(isObjectArrayControl, isPrimitiveArrayControl)
+  3,
+  or(isObjectArrayControl, isPrimitiveArrayControl)
 );
 
-class TableArrayControl extends React.Component<ArrayControlProps & VanillaRendererProps, any> {
-
+class TableArrayControl extends React.Component<
+  ArrayControlProps & VanillaRendererProps,
+  any
+> {
   render() {
     const {
       addItem,
@@ -90,8 +84,10 @@ class TableArrayControl extends React.Component<ArrayControlProps & VanillaRende
     const tableClass = getStyleAsClassName('array.table.table');
     const labelClass = getStyleAsClassName('array.table.label');
     const buttonClass = getStyleAsClassName('array.table.button');
-    const controlClass = [getStyleAsClassName('array.table'),
-      convertToValidClassName(controlElement.scope)].join(' ');
+    const controlClass = [
+      getStyleAsClassName('array.table'),
+      convertToValidClassName(controlElement.scope)
+    ].join(' ');
     const createControlElement = (key?: string): ControlElement => ({
       type: 'Control',
       label: false,
@@ -105,10 +101,11 @@ class TableArrayControl extends React.Component<ArrayControlProps & VanillaRende
     return (
       <div className={controlClass} hidden={!visible}>
         <header>
-          <label className={labelClass}>
-            {labelText}
-          </label>
-          <button className={buttonClass} onClick={addItem(path, createDefaultValue())}>
+          <label className={labelClass}>{labelText}</label>
+          <button
+            className={buttonClass}
+            onClick={addItem(path, createDefaultValue())}
+          >
             Add to {labelObject.text}
           </button>
         </header>
@@ -117,77 +114,91 @@ class TableArrayControl extends React.Component<ArrayControlProps & VanillaRende
         </div>
         <table className={tableClass}>
           <thead>
-          <tr>
-            {
-              schema.properties ?
+            <tr>
+              {schema.properties ? (
                 fpflow(
                   fpkeys,
                   fpfilter(prop => schema.properties[prop].type !== 'array'),
                   fpmap(prop => <th key={prop}>{prop}</th>)
-                ) (schema.properties)
-                : <th>Items</th>
-            }
-            <th>
-              Valid
-            </th>
-          </tr>
+                )(schema.properties)
+              ) : (
+                <th>Items</th>
+              )}
+              <th>Valid</th>
+            </tr>
           </thead>
           <tbody>
-          {
-            (!data || !Array.isArray(data) || data.length === 0) ?
-              <tr><td>No data</td></tr> :
+            {!data || !Array.isArray(data) || data.length === 0 ? (
+              <tr>
+                <td>No data</td>
+              </tr>
+            ) : (
               data.map((_child, index) => {
-                const childPath = Paths.compose(path, `${index}`);
+                const childPath = Paths.compose(
+                  path,
+                  `${index}`
+                );
                 // TODO
-                const errorsPerEntry: any[] = filter(
-                  childErrors,
-                  error => error.dataPath.startsWith(childPath)
+                const errorsPerEntry: any[] = filter(childErrors, error =>
+                  error.dataPath.startsWith(childPath)
                 );
 
                 return (
                   <tr key={childPath}>
-                    {
-                      schema.properties ?
-                        fpflow(
-                          fpkeys,
-                          fpfilter(prop => schema.properties[prop].type !== 'array'),
-                          fpmap(prop => {
-                            const childPropPath = Paths.compose(childPath, prop.toString());
+                    {schema.properties ? (
+                      fpflow(
+                        fpkeys,
+                        fpfilter(
+                          prop => schema.properties[prop].type !== 'array'
+                        ),
+                        fpmap(prop => {
+                          const childPropPath = Paths.compose(
+                            childPath,
+                            prop.toString()
+                          );
 
-                            return (
-                              <td key={childPropPath}>
-                                <DispatchField
-                                  schema={schema}
-                                  uischema={createControlElement(prop)}
-                                  path={childPath}
-                                  showError
-                                />
-                              </td>
-                            );
-                          })
-                        )(schema.properties) :
-                        <td key={Paths.compose(childPath, index.toString())}>
-                          <DispatchField
-                            schema={schema}
-                            uischema={createControlElement()}
-                            path={childPath}
-                            showError
-                          />
-                        </td>
-                    }
+                          return (
+                            <td key={childPropPath}>
+                              <DispatchField
+                                schema={schema}
+                                uischema={createControlElement(prop)}
+                                path={childPath}
+                              />
+                            </td>
+                          );
+                        })
+                      )(schema.properties)
+                    ) : (
+                      <td
+                        key={Paths.compose(
+                          childPath,
+                          index.toString()
+                        )}
+                      >
+                        <DispatchField
+                          schema={schema}
+                          uischema={createControlElement()}
+                          path={childPath}
+                        />
+                      </td>
+                    )}
                     <td>
-                      {
-                        errorsPerEntry ?
-                          <span className={getStyleAsClassName('array.validation.error')}>
-                            {join(errorsPerEntry.map(e => e.message), ' and ')}
-                          </span> :
-                          <span>OK</span>
-                      }
+                      {errorsPerEntry ? (
+                        <span
+                          className={getStyleAsClassName(
+                            'array.validation.error'
+                          )}
+                        >
+                          {join(errorsPerEntry.map(e => e.message), ' and ')}
+                        </span>
+                      ) : (
+                        <span>OK</span>
+                      )}
                     </td>
                   </tr>
                 );
               })
-          }
+            )}
           </tbody>
         </table>
       </div>
@@ -196,6 +207,6 @@ class TableArrayControl extends React.Component<ArrayControlProps & VanillaRende
 }
 
 export default connect(
-    addVanillaControlProps<StatePropsOfControl>(mapStateToArrayControlProps),
-    mapDispatchToArrayControlProps
+  addVanillaControlProps<StatePropsOfControl>(mapStateToArrayControlProps),
+  mapDispatchToArrayControlProps
 )(TableArrayControl);

--- a/packages/vanilla/src/complex/array/ArrayControl.tsx
+++ b/packages/vanilla/src/complex/array/ArrayControl.tsx
@@ -28,20 +28,17 @@ import { ArrayControlProps, composePaths } from '@jsonforms/core';
 import { ResolvedJsonForms } from '@jsonforms/react';
 import { VanillaRendererProps } from '../../index';
 
-export const ArrayControl  = (
-  {
-    classNames,
-    data,
-    label,
-    path,
-    schema,
-    addItem,
-    uischema,
-    findUISchema,
-    createDefaultValue
-  }: ArrayControlProps & VanillaRendererProps
-) => {
-
+export const ArrayControl = ({
+  classNames,
+  data,
+  label,
+  path,
+  schema,
+  addItem,
+  uischema,
+  findUISchema,
+  createDefaultValue
+}: ArrayControlProps & VanillaRendererProps) => {
   return (
     <div className={classNames.wrapper}>
       <fieldset className={classNames.fieldSet}>
@@ -52,14 +49,11 @@ export const ArrayControl  = (
           >
             +
           </button>
-          <label className={'array.label'}>
-            {label}
-          </label>
+          <label className={'array.label'}>{label}</label>
         </legend>
         <div className={classNames.children}>
-          {
-            data ? range(0, data.length).map(index => {
-
+          {data ? (
+            range(0, data.length).map(index => {
               const foundUISchema = findUISchema(schema, uischema.scope, path);
               const childPath = composePaths(path, `${index}`);
 
@@ -71,8 +65,10 @@ export const ArrayControl  = (
                   key={childPath}
                 />
               );
-            }) : <p>No data</p>
-          }
+            })
+          ) : (
+            <p>No data</p>
+          )}
         </div>
       </fieldset>
     </div>

--- a/packages/vanilla/src/controls/InputControl.tsx
+++ b/packages/vanilla/src/controls/InputControl.tsx
@@ -25,23 +25,27 @@
 import maxBy from 'lodash/maxBy';
 import React from 'react';
 import {
-    computeLabel, ControlProps,
-    ControlState,
-    formatErrorMessage,
-    isControl,
-    isDescriptionHidden,
-    isPlainLabel,
-    mapStateToControlProps,
-    NOT_APPLICABLE,
-    RankedTester,
-    rankWith,
+  computeLabel,
+  ControlProps,
+  ControlState,
+  formatErrorMessage,
+  isControl,
+  isDescriptionHidden,
+  isPlainLabel,
+  mapStateToControlProps,
+  NOT_APPLICABLE,
+  RankedTester,
+  rankWith
 } from '@jsonforms/core';
 import { Control, DispatchField } from '@jsonforms/react';
 import { addVanillaControlProps } from '../util';
 import { VanillaRendererProps } from '../index';
 import { connect } from 'react-redux';
 
-export class InputControl extends Control<ControlProps & VanillaRendererProps, ControlState> {
+export class InputControl extends Control<
+  ControlProps & VanillaRendererProps,
+  ControlState
+> {
   render() {
     const {
       classNames,
@@ -58,11 +62,20 @@ export class InputControl extends Control<ControlProps & VanillaRendererProps, C
     } = this.props;
 
     const isValid = errors.length === 0;
-    const divClassNames = `validation  ${isValid ? classNames.description : 'validation_error'}`;
-    const showDescription = !isDescriptionHidden(visible, description, this.state.isFocused);
+    const divClassNames = `validation  ${
+      isValid ? classNames.description : 'validation_error'
+    }`;
+    const showDescription = !isDescriptionHidden(
+      visible,
+      description,
+      this.state.isFocused
+    );
     const labelText = isPlainLabel(label) ? label : label.default;
     const field = maxBy(fields, r => r.tester(uischema, schema));
-    if (field === undefined || field.tester(uischema, schema) === NOT_APPLICABLE) {
+    if (
+      field === undefined ||
+      field.tester(uischema, schema) === NOT_APPLICABLE
+    ) {
       console.warn('No applicable field found.', uischema, schema);
       return null;
     } else {
@@ -78,14 +91,17 @@ export class InputControl extends Control<ControlProps & VanillaRendererProps, C
             {computeLabel(labelText, required)}
           </label>
           <DispatchField
-              uischema={uischema}
-              schema={schema}
-              path={parentPath}
-              id={id + '-input'}
-              showError={false}
+            uischema={uischema}
+            schema={schema}
+            path={parentPath}
+            id={id + '-input'}
           />
           <div className={divClassNames}>
-            {!isValid ? formatErrorMessage(errors) : showDescription ? description : null}
+            {!isValid
+              ? formatErrorMessage(errors)
+              : showDescription
+              ? description
+              : null}
           </div>
         </div>
       );

--- a/packages/vanilla/test/renderers/TableArrayControl.test.tsx
+++ b/packages/vanilla/test/renderers/TableArrayControl.test.tsx
@@ -132,9 +132,9 @@ test('render two children', t => {
 
   const tds = TestUtils.scryRenderedDOMComponentsWithTag(tree, 'td');
   t.is(tds.length, 3);
-  t.is(tds[0].children.length, 2);
+  t.is(tds[0].children.length, 1);
   t.is(tds[0].children[0].id, '');
-  t.is(tds[1].children.length, 2);
+  t.is(tds[1].children.length, 1);
   t.is(tds[1].children[0].id, '');
 });
 

